core/templates/domain/skill/skill-backend-api.service.ts

```diff
@@ -183,6 +187,34 @@ export class SkillBackendApiService {
       this._doesSkillWithDescriptionExist(description, resolve, reject);
     });
   }
+
+  private _getTopicNamesWithGivenSkillAssignedForDiagnosticTest(
```

Comment:
Why does this need to be broken out into a new function, rather than inlined in getTopicNamesWithGivenSkillAssignedForDiagnosticTest()?

I'm also not completely sure I understand it. The intermediate functions seem to have generalized success/error callbacks but the caller function doesn't even specify these. So can the logic here be simplified?

Reply:
Simplified with only a single function logic, please take a look.